### DISCUSSION TITLE
How to restore file backups on tshock ?

### DISCUSSION THREAD

**Discussion by hieu200219** [Pryaxis/TShock#2990] [Q&A]
Idk how to restore file backup on tshock pls help me i don’t see file world.wld

**Comment by sgkoishi** [comment]
Answer but I'll post here too: put it in the save directory. See https://web.archive.org/web/*/https://savelocation.net/terraria-character
~~So surprised there is a website that records the save locations~~

**Reply by hieu200219** [reply]
Can I ask one more question?
how to stop players from dupe items when the player dies he gets thrown items sometimes it will x2 items
and how to adjust last night's boss health in tshock

**Reply by sgkoishi** [reply]
* There's basically nothing you can do to stop item dupe/glitch/cheating.
* Some custom NPC or boss health plugin exists. You could also try to make your world journey mode so that you can use the slider to adjust in range.

**Reply by hieu200219** [reply]
I can't adjust directly in the master world? So do you know any plugins that can be adjusted? I've looked through all the plugins on github but still can't find it.

**Reply by ghost** [reply]
@sgkoishi said:
>* Some custom NPC or boss health plugin exists
--------------------------
Look for some @hieu200219 ...
